File: 1302deepestLeavesSum/deepestLeavesSum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int deepestLeavesSum(TreeNode root) {
        List<Integer> result = null;
        List<TreeNode> level = Arrays.asList(root);
        while (root != null && !level.isEmpty())
        {
            result = level.stream().map(node -> node.val).collect(Collectors.toList());
            level = level.stream().flatMap(node -> Stream.of(node.left, node.right).filter($ -> $ != null)).collect(Collectors.toList());
        }
        return result.stream().mapToInt(Integer::intValue).sum();
    }
}

File: 1302deepestLeavesSum/deepestLeavesSum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def deepestLeavesSum(self, root: TreeNode) -> int:
        result, level = None, [root]
        while root and level:
            result = (node.val for node in level)
            level = [child for node in level for child in (node.left, node.right) if child]
        return sum(result)
